### DISCUSSION TITLE
Can I add more python library myself

### DISCUSSION THREAD

**Discussion by pohchen** [exodrifter/unity-python#8] [Q&A]
Sorry for a question, how can i add more python library?
ex. Numpy, Scikit-learn

Need your help if you can guide me with any clues,
Many thanks for your help!

**Comment by exodrifter** [comment]
You are always welcome to make a fork and add your own libraries. Generally speaking, what I've done in the past is copy the python files from the system installation to the `Lib` folder.

I'm not sure if that's the best or proper way to do that though. If you learn more about how to do what you're asking, it would be nice if you could share so everyone can benefit.

**Comment by miryuu315315** [comment]
Hi. i'm japanese.
 i can little english.i 
can reading english but not writting. sorry...

↓try↓
C:\Users\user\Documents\ExanpleProject (8)\Assets\Python\Lib\site-packages\numpy
but can't use numpy.(Python2.7's numpy)

↓errormessage↓
ImportException: cannot import multiarray from numpy.core

↓C#code↓
```
using UnityEngine;
using IronPython.Hosting;
using Microsoft.Scripting.Hosting;

namespace Exodrifter.UnityPython.Examples
{
    public class PythonUnityHelloWorld : MonoBehaviour
    {
        void Start()
        {
            var engine = global::UnityPython.CreateEngine();
            var scope = engine.CreateScope();

            string code = "import UnityEngine\n";
            code += "UnityEngine.Debug.Log('Hello world!')\n";
            code += "import sys\n";

            code += "sys.path.append(r'C:\\Users\\user\\Documents\\ExanpleProject (8)\\Assets\\Python\\Lib\\site-packages')\n";
            code += "sys.path.append(r'C:\\Users\\user\\Documents\\ExanpleProject (8)\\Assets\\Python\\Lib')\n";
            code += "sys.path.append(r'C:\\Users\\user\\Documents\\ExanpleProject (8)\\Assets\\Python')\n";
            code += "sys.path.append(r'C:\\Users\\user\\Documents\\ExanpleProject (8)\\Assets\\Python\\IronPython - 2.7.8')\n";


            code += ("import numpy\n");
            code += "\n";

            var source = engine.CreateScriptSourceFromString(code);
            source.Execute(scope);
        }
    }
}
```
need help...

**Comment by offchan42** [comment]
After messing with adding `dlib` library into the system, I've found that it's impossible to import `.pyd` file using `IronPython`.
They suggest using `pythonnet` instead.
https://stackoverflow.com/questions/38425409/how-can-i-import-pyd-module-in-ironpython
So I think you should migrate from `IronPython` to `pythonnet`.

**Comment by exodrifter** [comment]
@miryuu315315 and @off99555: Please open a new issue instead of commenting on an irrelevant one.
